lib/datadog/core/backport.rb

```diff
@@ -21,5 +21,25 @@ def string_delete_prefix(string, prefix)
         end
       end
     end
+
+    # This module is used to provide features from Ruby 2.4+ to older Rubies
+    module BackportFrom24
+      module_function
+
+      if RUBY_VERSION < '2.4'
+        def dup(value)
+          case value
+          when NilClass, TrueClass, FalseClass, Integer, Float
```

Comment:
I'm not sure if it's worth caring, but out of pure curiosity I went and checked and `Bignum` (e.g. `2**300.dup`) also breaks on 2.3 and works on modern Rubies.

Reply:
Yeah, I thought about adding it as well. But I wasn't sure if we use `Bignum` on our codebase. I'm guessing not, but adding it just in case makes sense to avoid future 💥

Reply:
Yeap I vaguely remembered that. Yay for modern rubies!